### PR TITLE
Add ROIAlign pooling

### PR DESCRIPTION
**Pull request by JihongJu**
Using `tensorflow.image.crop_and_resize`. 
`crop_and_resize` uses the bounding boxes defined by the normalized coordinates of the top left and bottom right corners [y1, x1, y2, x2] instead of [x, y, w, h].

As proposed in broadinstitute/keras-rcnn#16.

### CODE REVIEW

File: keras_rcnn/layers/pooling.py [hunk 1]
```diff
@@ -55,3 +57,48 @@ def call(self, x, **kwargs):
...
+
+        # normalize the boxes
+        shape = image._keras_shape
+        h = keras.backend.cast(shape[1], keras.backend.floatx())
+        w = keras.backend.cast(shape[2], keras.backend.floatx())
+        x1 /= w
+        y1 /= h
+        x2 /= w
+        y2 /= h
+        boxes = keras.backend.concatenate([y1[:, numpy.newaxis],
```
Comment (0x00b1) [inline keras_rcnn/layers/pooling.py]: Can you use `numpy.newaxis` here? I’d think you’d need to use `tensorflow.newaxis`.

File: keras_rcnn/layers/pooling.py [hunk 2]
```diff
@@ -55,3 +57,48 @@ def call(self, x, **kwargs):
 
     def compute_output_shape(self, input_shape):
         return None, self.regions, self.size, self.size, self.channels
+
+
+class ROIAlign(ROI):
```
Comment (0x00b1) [inline keras_rcnn/layers/pooling.py]: I like your implementation. It reminds me of the G-RMI implementation from Jonathan Huang, et al.

File: keras_rcnn/layers/pooling.py [hunk 3]
```diff
@@ -55,3 +57,48 @@ def call(self, x, **kwargs):
...
+        # convert regions from (x, y, w, h) to (x1, y1, x2, y2)
+        regions = keras.backend.cast(regions, keras.backend.floatx())
+        regions = regions / (self.stride + 0.0)
+        x1 = regions[:, 0]
+        y1 = regions[:, 1]
+        x2 = regions[:, 0] + regions[:, 2]
+        y2 = regions[:, 1] + regions[:, 3]
+
+        # normalize the boxes
+        shape = image._keras_shape
```
Comment (0x00b1) [inline keras_rcnn/layers/pooling.py]: `_keras_shape` is private. I think you should use `keras.backend.int_shape`.

File: keras_rcnn/layers/pooling.py [hunk 4]
```diff
@@ -55,3 +57,48 @@ def call(self, x, **kwargs):
...
+        super(ROIAlign, self).__init__(size=size,
+                                       regions=regions,
+                                       **kwargs)
+
+    def call(self, x, **kwargs):
+        image, regions = x[0], x[1]
+
+        # convert regions from (x, y, w, h) to (x1, y1, x2, y2)
+        regions = keras.backend.cast(regions, keras.backend.floatx())
+        regions = regions / (self.stride + 0.0)
```
Comment (0x00b1) [inline keras_rcnn/layers/pooling.py]: Shouldn’t `0.0` be `keras.backend.epsilon()`?

Comment (JihongJu) [inline keras_rcnn/layers/pooling.py]: @0x00b1 No basically what happens here is to cast the stride to a float. So probably `keras.backend.cast(self.stride, keras.backend.floatx())` is better.

File: keras_rcnn/layers/pooling.py (hunk 1 above)
Comment (JihongJu) [inline keras_rcnn/layers/pooling.py]: Unfortunately, I don't know neither. But I tried to keep the `tensorflow.foo` apart from the layer implementation. Does `tensorflow.neaxis` work with theano backend?

**Comment by JihongJu** [comment]
@0x00b1 Maybe review again?

**Comment by 0x00b1** [comment]
@JihongJu How would you feel about replacing ROI with ROIAlign?

**Comment by JihongJu** [comment]
@0x00b1 I agree with you. According to Mask R-CNN (He et. al.), ROIAlign "improves AP by ∼3 points and
AP75 by ∼5 point". And I think this is significant enough.

**Comment by 0x00b1** [comment]
@JihongJu Totally. I guess you collapse ROIAlign into the ROI class.

**Comment by JihongJu** [comment]
@0x00b1 Done

**Comment by 0x00b1** [comment]
LGTM
